src/ol/style/regularshape.js

```diff
@@ -401,11 +431,12 @@ ol.style.RegularShape.prototype.draw_ = function(renderOptions, context, x, y) {
   context.translate(x, y);
 
   context.beginPath();
+  var points = this.points_;
   if (this.radius2_ !== this.radius_) {
-    this.points_ = 2 * this.points_;
+    points = 2 * points;
   }
-  for (i = 0; i <= this.points_; i++) {
-    angle0 = i * 2 * Math.PI / this.points_ - Math.PI / 2 + this.angle_;
+  for (i = 0; i <= points; i++) {
+    angle0 = i * 2 * Math.PI / points - Math.PI / 2 + this.angle_;
```

Comment:
i introduced a new variable here to store the doubled points, else the value would be doubled after each clone (and each call of draw_ and drawHitDetectionCanvas_).